### PR TITLE
Add lazy loading and async decoding to images

### PR DESCRIPTION
**Pull request by mohammadareeb95**
Towards: https://github.com/wagtail/sphinx_wagtail_theme/issues/177

### CODE REVIEW

File: sphinx_wagtail_theme/layout.html [hunk 1]
```diff
@@ -89,7 +89,7 @@
                             {%- if theme_logo_height %} height="{{ theme_logo_height|e }}"{% endif %}
                             {%- if theme_logo_alt %} alt="{{ theme_logo_alt|e }}"{% endif %}
                             {%- if theme_logo_title %} title="{{ theme_logo_title|e }}"{% endif %}
-                            class="logo-img"
+                            class="logo-img" decoding="async" loading="lazy"
```
Comment (lb-) [inline sphinx_wagtail_theme/layout.html:92]: nitpick - please put these new attributes on their own lines - as all other attributes are on their own lines.

Also - maybe this should show instantly as it's the primary logo for the docs.

File: sphinx_wagtail_theme/domainindex.html [hunk 2]
```diff
@@ -40,7 +40,7 @@ <h1>{{ indextitle }}</h1>
      <tr{% if grouptype == 2 %} class="cg-{{ groupid.current() }}"{% endif %}>
        <td>{% if grouptype == 1 -%}
          <img src="{{ pathto('_static/minus.png', 1) }}" class="toggler"
-              id="toggle-{{ groupid.next() }}" style="display: none" alt="-" />
+              id="toggle-{{ groupid.next() }}" style="display: none" alt="-" decoding="async" loading="lazy" />
```
Comment (lb-) [inline sphinx_wagtail_theme/domainindex.html:43]: I think this image should not be async lazy as we need this to show instantly on load as it's a menu button.

**Review by lb- — COMMENTED** [review]
We might need to add some nuance to the issue, the lazy images should really only be images that are in the documentation pages, not core images for the docs to show on first load (e.g. header logo / menu toggle - should be visible straight away).

I think these changes will probably not be what we want sorry.

File: sphinx_wagtail_theme/layout.html (hunk 1 above)
Comment (mohammadareeb95) [reply sphinx_wagtail_theme/layout.html:92]: sure!

**Pull request closed without merging**